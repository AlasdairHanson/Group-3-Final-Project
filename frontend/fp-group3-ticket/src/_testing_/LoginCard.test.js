import {BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';
import LoginCard from '../Components/LandingPage/LoginCard';


  describe(`Test of Login Card`, () => {

    test('Renders Log In', () => {
        render(<Router><LoginCard /></Router>);
        const linkElement = screen.getByText(/log in/i);
        expect(linkElement).toBeInTheDocument();
      });

    test('Renders Email Address', () => {
        render(<Router><LoginCard /></Router>);
        const linkElement = screen.getByText(/email address/i);
        expect(linkElement).toBeInTheDocument();
      });

    test('Renders Password', () => {
        render(<Router><LoginCard /></Router>);
        const linkElement = screen.getByText(/password/i);
        expect(linkElement).toBeInTheDocument();
      });

    test(` Login Card should match the snapshot every time it is rendered`, () => {
      const instance = create(<Router><LoginCard/></Router>).toJSON(); 
      expect(instance).toMatchSnapshot();
    });
  });