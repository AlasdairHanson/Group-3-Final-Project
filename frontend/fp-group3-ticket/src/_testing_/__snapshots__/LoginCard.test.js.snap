// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Login Card  Login Card should match the snapshot every time it is rendered 1`] = `
<div
  className="loginCard card"
>
  <div
    className="card-body"
  >
    <div
      className="text-center card-title h5"
    >
      <h3>
        Log in
      </h3>
    </div>
    <div
      className="form-group"
    >
      <label>
        Email address
      </label>
      <input
        className="form-control"
        placeholder="Enter email"
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
      </label>
      <input
        className="form-control"
        placeholder="Enter password"
        type="password"
      />
    </div>
    <a
      href="/tickets"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary btn-block"
        type="submit"
      >
        Submit
      </button>
    </a>
  </div>
</div>
`;
